File: admin-lingopro/src/models/Lesson.ts
import { Timestamp } from "firebase/firestore";

export type LessonStatus = "active" | "inactive" | "deleted";

export interface Lesson {
  lessonId: string;
  topicId: string;
  topicName?: string; // Tên topic để hiển thị trong bảng
  title: string;
  description: string;
  content: string;
  order: number;
  status: LessonStatus;
  createdAt: Timestamp;
  updatedAt: Timestamp;
  createdBy: string;
  updatedBy: string;
}

File: app/src/main/java/com/example/englishlearningapp/data/model/Lesson.kt
package com.example.englishlearningapp.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class Lesson(
    val lessonId: String = "",
    val topicId: String = "",
    val name: String = "",
    val description: String = "",
    val order: Int = 0,
    val totalWords: Int = 0,
    val status: String = "active",
    val createdAt: Timestamp? = null,
    val updatedAt: Timestamp? = null,
    val createdBy: String = "",
    val updatedBy: String = "",
    val content: String = ""
)
